crates/oxc_linter/src/rules/react/rules_of_hooks.rs

```diff
@@ -172,13 +172,12 @@ impl Rule for RulesOfHooks {
             }) => {
                 let ident = get_declaration_identifier(nodes, parent_func.id());
 
-                // Hooks cannot be called inside of export default functions or used in a function
-                // declaration outside of a react component or hook.
+                // Hooks cannot be used in a function declaration outside of a react component or hook.
```

Comment:
cc @rzvxa looks like you added this comment originally. but, as far as i understand there is nothing incorrect about calling a hook within a function that's default exported + react doesn't report

let me know if there was a reason this should exist - thanks!